runtime/vm/createramclass.cpp

```diff
@@ -1571,22 +1580,28 @@ loadSuperClassAndInterfaces(J9VMThread *vmThread, J9ClassLoader *classLoader, J9
 			}
 
 			/* ensure that the superclass is visible */
-			if (!isROMClassUnsafe) {
+			if (!isExemptFromValidation) {
 				/*
 				 * Failure occurs if
-				 * 1) The superClass class not public and does not belong to
+				 * 1) The superClass class is not public and does not belong to
 				 * the same package as the class being loaded.
 				 * 2) The superClass class is public but the class being loaded
 				 * belongs to a module that doesn't have access to the module that
 				 * owns the superClass class.
 				 */
 				bool superClassIsPublic = J9_ARE_ALL_BITS_SET(superclass->romClass->modifiers, J9AccPublic);
-				if ((!superClassIsPublic && (packageID != superclass->packageID))
-					|| (superClassIsPublic && (J9_VISIBILITY_ALLOWED != checkModuleAccess(vmThread, vm, romClass, module, superclass->romClass, superclass->module, superclass->packageID, 0)))
+				if (superClassIsPublic
+						? (J9_VISIBILITY_ALLOWED != checkModuleAccess(vmThread, vm, romClass, module, superclass->romClass, superclass->module, superclass->packageID, 0))
```

Comment:
Please revert this refactoring - embedding a ternary if inside an if statement is painful from a readability perspective.  Repeating the single boolean check is fine if the compiler isn't smart enough to remove it

Reply:
I disagree: I find it clearer and easier to understand this way. I will revert if you insist.

Reply:
I will revert this change (and the similar one below): we can return to this discussion later.